File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()<2){
            return nums[0];
        }
        int globalsum=nums[0];
        int localsum=nums[0];
        for(int i=1; i<nums.size(); i++){
            localsum  = max(nums[i], localsum+ nums[i]);
            globalsum=max(globalsum,localsum);
        }
        return globalsum;

    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int i = 0;
        int j = 0;
        int sum =0;
        int ans = Integer.MIN_VALUE;
        while (j<nums.length && i < nums.length){
            while (sum<0){
                sum -=nums[i];
                i++;
            }
            sum+=nums[j];
            ans = Math.max(ans, sum);
            j++;
        }


        return ans;
    }
}